packages/design-system/src/components/OcFileInput/OcFileInput.vue

```diff
@@ -0,0 +1,262 @@
+
+defineOptions({
+  inheritAttrs: false
+})
+
+export interface Props {
+  /**
+   * @docs The label of the input element.
+   */
+  label: string
+  /**
+   * @docs The element ID of the input.
+   */
+  id?: string
+  /**
+   * @docs The file types that the input should accept. E.g. 'image/*, .pdf'.
+   * @default ''
+   */
+  fileTypes?: string
+  /**
+   * @docs Determines if the input should accept multiple files.
+   * @default false
+   */
+  multiple?: boolean
+  /**
+   * @docs Determines if the input should have a clear button. Only gets displayed if the input has a value.
+   * @default true
+   */
+  clearButtonEnabled?: boolean
+  /**
+   * @docs Determines if the input is disabled.
+   * @default false
+   */
+  disabled?: boolean
+  /**
+   * @docs The error message to be displayed below the input.
+   */
+  errorMessage?: string
+  /**
+   * @docs Determines if the message line should be fixed.
+   * @default false
+   */
+  fixMessageLine?: boolean
+  /**
+   * @docs The description message to be displayed below the input.
+   */
+  descriptionMessage?: string
+  /**
+   * @docs Determines if a required mark (*) should be displayed next to the label.
+   * @default false
+   */
+  requiredMark?: boolean
+}
+
+export interface Emits {
+  /**
+   * @docs Emitted when the value of the input has updated.
+   */
+  (e: 'update:modelValue', value: FileList): void
+  /**
+   * @docs Emitted when the input has been focused.
+   */
+  (e: 'focus', value: HTMLElement): void
+}
+
+export interface Slots {
+  /**
+   * @docs Can be used to overwrite the default rendering of the label.
+   */
+  label?: () => unknown
+}
+
+const {
+  label,
+  id = uniqueId('oc-fileinput-'),
+  fileTypes = '',
+  multiple = false,
+  clearButtonEnabled = true,
+  disabled = false,
+  errorMessage = '',
+  fixMessageLine = false,
+  descriptionMessage = '',
+  requiredMark = false
+} = defineProps<Props>()
+
+const emit = defineEmits<Emits>()
+defineSlots<Slots>()
+
+const inputValue = ref('')
+
+const showMessageLine = computed(() => {
+  return fixMessageLine || !!errorMessage || !!descriptionMessage
+})
+
+const messageId = computed(() => `${id}-message`)
+
+const tmpAttrs = useAttrs()
+const additionalAttributes = computed(() => {
+  const additionalAttrs: Record<string, unknown> = {}
+  if (!!errorMessage || !!descriptionMessage) {
+    additionalAttrs['aria-describedby'] = messageId.value
+  }
+
+  return { ...tmpAttrs, ...additionalAttrs }
+})
+
+const ariaInvalid = computed(() => {
+  return (!!errorMessage).toString()
+})
+
+const messageText = computed(() => {
+  if (errorMessage) {
+    return errorMessage
+  }
+  return descriptionMessage
+})
+
+const inputRef = useTemplateRef<HTMLInputElement>('inputRef')
+const inputBtnRef = useTemplateRef<HTMLElement>('inputBtnRef')
+
+const focus = () => {
+  unref(inputRef).focus()
+}
+defineExpose({ focus })
+
+const fileNames = computed(() => {
+  if (unref(inputRef)?.files) {
```

Comment:
This computed doesn't get re-evaluated:
- when I clear the selected file and add another selected file
- when I don't clear the selected file but just select another file when a file was already selected before